src/mongoose_redis.erl

```diff
@@ -0,0 +1,65 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%%
+%%% @end
+%%% Created : 27. Jun 2018 17:16
+%%%-------------------------------------------------------------------
+-module(mongoose_redis).
+-author("bartlomiej.gorny@erlang-solutions.com").
+-include("mongoose.hrl").
+-define(POOL_NAME, redis_pool).
+
+%% API
+-export([start_pool/1, cmd/1, cmd/2]).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec start_pool(list()) -> {ok, pid()} | {error, {already_started, pid()}}.
+start_pool(Opts) ->
+    mongoose_wpool:setup_env(),
+    PoolSize = proplists:get_value(pool_size, Opts, 10),
+    RedisOpts = proplists:get_value(worker_config, Opts, []),
+    PoolOpts = proplists:get_value(pool_opts, Opts, []),
+    PoolOptions = [{workers, PoolSize},
+                   {worker, {eredis_client, makeargs(RedisOpts)}}
+                  ] ++ PoolOpts,
+    Res = wpool:start_pool(?POOL_NAME, PoolOptions),
+    case Res of
+        {ok, _Pid} -> Res;
+        Error ->
+            ?ERROR_MSG("Failed to start worker pool, reason: ~p~n", [Error]),
+            Error
+    end.
+
+-spec cmd(iolist()) -> undefined
+                       | binary()
+                       | [binary() | [binary() | integer()] | integer() | {'error', _}]
+                       | integer()
+                       | {'error', _}.
+cmd(Cmd) ->
+    cmd(Cmd, 5000).
+
+-spec cmd(iolist(), integer()) -> undefined
+                                  | binary()
+                                  | [binary() | [binary() | integer()] | integer() | {'error', _}]
+                                  | integer()
+                                  | {'error', _}.
+cmd(Cmd, Timeout) ->
+    case eredis:q(wpool_pool:random_worker(?POOL_NAME), Cmd, Timeout) of
```

Comment:
I'd consider using `next_worker` (the same behaviour as cuesport) or `next_available_worker` (this would require some more work to handle the exception properly).

Reply:
I suggested this one because of what I read in the documentation: "This strategy is the fastest one when to select a worker. It's ideal if your workers will perform many short tasks." - which seems to be the case here.

Reply:
It doesn't really matter if it is random_worker and next_worker, they boil down to pretty much the same. My understanding of available_worker strategy is that it is designed for situation where one operation can take really, really long, and if you are unlucky enough to pick the wrong worker then you can get stuck for several seconds or even minutes. Picking an available worker is much more expensive than simply selecting next or random, but it pays off it a sense that you avoid such risk. These redis queries or pipelines do not seem to fall into this category, "a bit longer" still means a tiny fraction of a second. So, can be next_worker, fine, but I wouldn't use available worker (or next available worker) here.

Reply:
All right, that's fine. Let's use the random worker as it is now.